libr/core/cmd_search.c

```diff
@@ -1600,8 +1611,18 @@ static void do_esil_search(RCore *core, struct search_parameters *param, const c
 	const bool cfgDebug = r_config_get_i (core->config, "cfg.debug");
 	RSearch *search = core->search;
 	RSearchKeyword kw = R_EMPTY;
-	if (input[0] == 'E' && input[1] != ' ') {
-		eprintf ("Usage: /E [esil-expr]\n");
+	if (input[0] != 'E') {
+		return;
+	}
+	PJ *pj = NULL;
+	if (input[1] == 'j') { // "/Ej"
+		// BUGGY and dupe not using pj, param->outmode = R_MODE_JSON;
+		pj = pj_new ();
+		pj_a (pj);
+		input++;
+	}
+	if (input[1] != ' ') { // "/E?"
```

Comment:
this should be an `else if`, no?

Reply:
no, because im doing input++ on purpose on the if above to make this if work as intented

Reply:
ok right